orderer/common/cluster/deliver_test.go

```diff
@@ -654,19 +654,23 @@ func TestBlockPullerFailover(t *testing.T) {
 
 	// Configure the block puller logger to signal the wait group once the block puller
 	// received the first block.
-	var pulledBlock1 sync.WaitGroup
-	pulledBlock1.Add(1)
+	pulledBlock1Done := make(chan struct{})
 	var once sync.Once
 	bp.Logger = bp.Logger.WithOptions(zap.Hooks(func(entry zapcore.Entry) error {
 		if strings.Contains(entry.Message, "Got block [1] of size") {
-			once.Do(pulledBlock1.Done)
+			once.Do(func() { close(pulledBlock1Done) })
 		}
 		return nil
 	}))
 
 	go func() {
 		// Wait for the block puller to pull the first block
-		pulledBlock1.Wait()
+		select {
+		case <-pulledBlock1Done:
+		case <-time.After(10 * time.Second):
```

Comment:
Did you choose 10 seconds as it really should never take longer than that?  I imagine when the test "works", it should be pretty quick?

Reply:
When the tests work, they take under a second so I went with something an order of magnitude longer. If the times need to change, it will be easy enough to see.

The real goal is to catch the source of the hang in one of the `TestBlockPullerBadBlocks` sub-tests and try to fix it.